desk/app/grouper.hoon

```diff
@@ -36,39 +36,50 @@
   ^-  (quip card _this)
   ?+  mark  (on-poke:def mark vase)
     %leave  :_  this  ~[[%pass /bite-wire %agent [our.bowl %reel] %leave ~]]
-      %watch  
-    :_  this  
+  ::
+      %watch
+    :_  this
     ?:  (~(has by wex.bowl) [/bite-wire our.bowl %reel])  ~
     ~[(bite-subscribe bowl)]
-    ::
+  ::
       %grouper-enable
     =+  !<(name=cord vase)
     `this(enabled-groups (~(put in enabled-groups) name))
+  ::
       %grouper-disable
     =+  !<(name=cord vase)
     `this(enabled-groups (~(del in enabled-groups) name))
+  ::
       %grouper-ask-enabled
     =+  !<(name=cord vase)
     =/  enabled  (~(has in enabled-groups) name)
     :_  this
     ~[[%pass [%ask name ~] %agent [src.bowl %grouper] %poke %grouper-answer-enabled !>([name enabled])]]
+  ::
       %grouper-answer-enabled
     =/  [name=cord enabled=?]  !<([cord ?] vase)
     :_  this
     ~[[%give %fact ~[[%group-enabled (scot %p src.bowl) name ~]] %json !>(b+enabled)]]
+  ::
       %grouper-check-link
-    =+  !<(=path vase)
-    ?>  ?=([%check-link @ @ ~] path)
+    =+  !<(=(pole knot) vase)
+    ?>  ?=([%check-link rest=*] pole)
     =/  baseurl  .^(cord %gx /(scot %p our.bowl)/reel/(scot %da now.bowl)/service/noun)
-    =/  target=ship  (slav %p i.t.path)
-    =/  group=cord  i.t.t.path
+    ::  it really is necessary to double-encode this, make sure we strip
+    ::  the leading slash before encoding
+    =/  end  (en-urlt:html (en-urlt:html +:(spud rest.pole)))
+    =/  url
+      ?.  =(baseurl 'https://tlon.network/lure/')
+        (crip "{(trip baseurl)}{end}")
+      (crip "https://tlon.network/v1/policies/lure/{end}")
```

Comment:
Here and in urbit/landscape#280, it's surprising to me that the base url is expected to be given with a trailing slash, leading you into the annoying pattern of needing to strip the leading slash off the "actual" path... Unclear if that's worth correcting now, or punting to later.

(Also I maintain that double-encoding (or needing to encode at all for that matter) is a smell, but I get that it's finicky so we can just hold our noses for now.)

Reply:
yeah it's bad just going to hold nose until we make a simpler lure service